From dbee03b95e83e8a3251074b2947dfc137656b8c8 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 15:30:14 -0400
Subject: [block] mtip32xx: Remove dfs_parent after pci unregister

Message-id: <1402673421-17117-12-git-send-email-snitzer@redhat.com>
Patchwork-id: 83922
O-Subject: [RHEL7.1 PATCH 11/18] mtip32xx: Remove dfs_parent after pci unregister
Bugzilla: 1105703
RH-Acked-by: David Milburn <dmilburn@redhat.com>

BZ: 1105703

Upstream commit af5ded8ccf21627f9614afc03b356712666ed225
Author: Asai Thambi S P <asamymuthupa@micron.com>
Date:   Wed Apr 16 20:30:16 2014 -0700

    mtip32xx: Remove dfs_parent after pci unregister

    In module exit, dfs_parent and it's subtree were removed before
    unregistering with pci. When debugfs entry for each device is attempted
    to remove in pci_remove() context, they don't exist, as dfs_parent and
    its children were already ripped apart.

    Modified to first unregister with pci and then remove dfs_parent.

    Signed-off-by: Asai Thambi S P <asamymuthupa@micron.com>
    Cc: stable@kernel.org
    Signed-off-by: Jens Axboe <axboe@fb.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/block/mtip32xx/mtip32xx.c b/drivers/block/mtip32xx/mtip32xx.c
index 9c22c3c..2942f40 100644
--- a/drivers/block/mtip32xx/mtip32xx.c
+++ b/drivers/block/mtip32xx/mtip32xx.c
@@ -4936,13 +4936,13 @@ static int __init mtip_init(void)
  */
 static void __exit mtip_exit(void)
 {
- debugfs_remove_recursive(dfs_parent);
-
  /* Release the allocated major block device number. */
  unregister_blkdev(mtip_major, MTIP_DRV_NAME);
 
  /* Unregister the PCI driver. */
  pci_unregister_driver(&mtip_pci_driver);
+
+ debugfs_remove_recursive(dfs_parent);
 }
 
 MODULE_AUTHOR("Micron Technology, Inc");
-- 
1.7.1